src/data_race.rs

```diff
@@ -681,7 +760,7 @@ impl VClockAlloc {
                         .enumerate()
                         .find_map(|(idx, &r)| if r == 0 { None } else { Some(idx) })
                         .expect("Invalid VClock Invariant");
-                    Some(idx)
+                    Some(idx + r_slice.len())
```

Comment:
Is this a bugfix, or what does this do?

Reply:
Bugfix, would return the incorrect vector index as the source of the data-race for VClocks with different sizes.

a race between [1,2] and [1] would return that the race was with index 0 instead of 1.

Reply:
Okay, thanks. Is there a test ensuring that this will not regress?

Reply:
Yes, i added all thread ids to the tests instead of matching on just //~ERROR data race.

The test dealloc_read_race_stack_drop.rs has this case occur and covers the bug by requiring explicit matching on
```
//~ ERROR Data race detected between Deallocate on Thread(id = 1) and Read on Thread(id = 2)
```
whereas with the bug it would return:
```
//~ ERROR Data race detected between Deallocate on Thread(id = 1) and Read on Thread(id = 0, name = "main") 
```

Reply:
Okay, awesome. :)